packages/run-protocol/src/vpool-xyk-amm/addPool.js

```diff
@@ -164,6 +160,13 @@ export const makeAddPoolInvitation = (
     const [liquidityKeyword] = Object.entries(zcf.getTerms().issuers).find(
       ([_, i]) => i === issuer,
     );
+    assert(liquidityKeyword, 'liquidity brand required');
+
+    // COMMIT POINT /////////////////////
```

Comment:
I like this convention. Aligns with https://github.com/Agoric/agoric-sdk/wiki/Prepare-Commit that @dckc started.

That says that the commit phase "must" be atomic. Should this all get wrapped in a try/catch? 

A convention that could help us to be more consistent and which would be easy to lint: define a `commit` function (could be within the handler so it doesn't need params) and in the end of the handler function,
```
try {
  await commit();
} catch {
  // abort so partial state is not observable
}
return result;
```
I don't know if that's the solution but if you think it would be feasible and worthwhile to have some more consistent structure I'd like to include it in https://github.com/Agoric/agoric-sdk/issues/4770

That's not for MN-1 but maybe an MN-1 thing would be to come up with some rules that we manually lint before release. Worth a ticket?

Reply:
> Aligns with Prepare-Commit that @dckc started.

well, at the request of @erights 

> Should this all get wrapped in a try/catch?

I think we avoid that as a rule. I haven't looked at the rest of the code in this case. Is there some reason to believe that the stuff after the commit-point not atomic?

> ... define a commit function

I think @erights tried that and the resulting reduction in clarity was judged to be a net loss. Here's hoping I/we can find that experiment. Oh... wait... he tried a utility function that served as the try-catch idiom... I think what you're suggesting here is a `commit()` function for each case.

We wouldn't want to `await` the `commit()`. The `commit()` should be in the same turn as when the prerequisites are checked.